### ISSUE TITLE
Copy variables not really working well

### ISSUE THREAD

**Issue by flip111** [felixfbecker/vscode-php-debug#400]
In the "Variables" pane where all the variables are listed in local scope (and a bunch of other scopes), when i have an array there and right click `Copy Value` i get

```
array(42)
```

When i click `Copy as Expression` i get

```
$forum_data_ary
```

These two options are somewhat useful. The "copy as expression" is more useful than "copy as value" though. Instead of just these two options could the following there be the following entries in the right click menu:

```
Copy as PHP (copies runnable PHP code to construct the variable)
Copy as JSON (copies JSON representation of variables)
Copy name (new name for "copy as expression")
Copy type  (new name for "copy as value")
```

**Comment by zobo** [comment]
Related to #917 but does not solve the issue, as currently one one option is possible, not a list of options. Opened an issue with VSCode.

**Comment by zobo** [comment]
For reference, the needed feature on VSCode side was not implemented https://github.com/microsoft/vscode/issues/188604 so I can't yet implement this.

**Comment by flip111** [comment]
@zobo thanks for looking into this!

I saw in the linked issue

> Yeah, that is not exposed right now. I guess it should be on that context object, or exposed as in https://github.com/microsoft/vscode/issues/63943

And in the referred issue (7 may 2024):

> This API has been finalized and will in tomorrow's Insiders and the next stable release. Please refer to https://github.com/microsoft/vscode/pull/212190/files#diff-0a75aed19c118603eb96332bc0b9c2d7867f4182346d16d18b7fc31b6ceeb321 for the final API.

Is that enough to implement it or is more needed?

**Comment by zobo** [comment]
No. The linked issue is talking about threads and stacks. This are elements in "CALL STACK" window. I'd need something similar for the "VARIABLES" windows. Only then could I implement extra menu items for all "Copy as ..." variants.

I'm going to finish implementing #917 one of this days, that will make a tree traversal when copying a variable with sub nodes and present it like "var_export"...

For my reference: I just found https://github.com/microsoft/vscode/blob/main/src/vscode-dts/vscode.proposed.debugVisualization.d.ts but that would probably not be a good solution also...

Semi unrelated note: Have you seen the new "debug.showVariableTypes" ?

**Comment by flip111** [comment]
> Have you seen the new "debug.showVariableTypes" ?

No i hadn't seen it yet. Reference: https://code.visualstudio.com/updates/v1_92#_show-variable-types-while-debugging

**Comment by flip111** [comment]
@ljubadr @cDonut @dracula-property @m10 @sylver @shaneparsons i saw that you upvoted this issue, thanks for that!

However it's blocked on this https://github.com/microsoft/vscode/issues/188604 which got [closed](https://github.com/microsoft/vscode/wiki/Issues-Triaging#our-triaging-flow) because not enough upvotes. Perhaps with [20 votes](https://github.com/microsoft/vscode/wiki/Issues-Triaging#up-voting-a-feature-request) we can ask a mod to reopen this ticket. Please see my first post how to upvote the ticket
